src/main/java/com/bootcamp/demo/service/BookingService.java

```diff
@@ -0,0 +1,166 @@
+                .delete()
+                .get()
+                .getUpdateTime()
+                .toString();
+    }
+
+    @Override
+    public String updateBooking(String bookingId, String endDate, PaymentStatus newStatus) throws ExecutionException, InterruptedException {
+        if (bookingId == null) throw new IllegalArgumentException("Parameter 'bookingId' cannot be null");
+        return db.collection(COLLECTION_PATH)
+                .document(bookingId)
+                .update("endDate", endDate, "paymentStatus", newStatus)
+                .get()
+                .getUpdateTime()
+                .toString();
+    }
+
+    public Scooter bookAScooter(final Double lat, final Double longitude) throws ExecutionException, InterruptedException {
+        Random random = new Random();
+        Location userLocation = new Location(lat, longitude);
+        Set<Scooter> allAvailableScooters = scooterService.getAvailableScooters(userLocation, 10.0);
+        if (allAvailableScooters.size() > 0) {
+            int randomIndexScooter = random.nextInt(allAvailableScooters.size());
+            Scooter[] listOfScooters = allAvailableScooters.toArray(new Scooter[allAvailableScooters.size()]);
+            Scooter pickedScooter = listOfScooters[randomIndexScooter];
+            pickedScooter.setStatus(ScooterStatus.RESERVED);
+            scooterService.updateScooter(pickedScooter.getSerialNumber(), pickedScooter.getCurrentLocation(), pickedScooter.getStatus(), pickedScooter.getBattery().getLevel());
+            LOGGER.info("scooter:" + pickedScooter.toString());
+            LinkedHashSet<Booking> bookings = getAllBookings();
+            int randomIndexBooking = random.nextInt(bookings.size());
+            Booking[] listOfBookings = bookings.toArray(new Booking[bookings.size()]);
+            Booking pickedBooking = listOfBookings[randomIndexBooking];
+            pickedBooking.setSerialNumber(pickedScooter.getSerialNumber());
+            String startDateBooking = pickedBooking.getStartDate();
+            pickedBooking.setStartDate(startDateBooking);
+            LOGGER.info("booking:" + pickedBooking.toString());
+            createBooking(pickedBooking);
+
+            return pickedScooter;
+        }
+        return null;
+    }
+
+    public Double endBookingAndUpdate(Booking booking, String endDate) throws ExecutionException, InterruptedException {
+        long bookingDuration = 0;
+        Scooter scooter = scooterService.findScooterById(booking.getSerialNumber());
+        String startDate = booking.getStartDate();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+        LocalDateTime dateTimeStart = LocalDateTime.parse(startDate, formatter);
+        LocalDateTime dateTimeEnd = LocalDateTime.parse(endDate, formatter);
+        if (dateTimeEnd.isAfter(dateTimeStart)) {
+            bookingDuration = dateTimeEnd.getMinute() - dateTimeStart.getMinute();

```

Comment:
Will this computation work as expected for all scenarios i.e. dateTimeEnd minute is smaller than dateTimeStart minute?
You could find other methods to substract dates and get the result in minutes (hint: Duration class)

Reply:
Fixed. I used both toMinutes and ofMinutes methods from Duration class.